function a(o,t){return()=>{try{const r=o();return r instanceof Promise?r.then(e=>[e,null]).catch(e=>{const n=t||"An error occurred during operation";return[null,new Error(`${n}: ${e instanceof Error?e.message:String(e)}`,{cause:e})]}):[r,null]}catch(r){const e="An error occurred during operation";return[null,new Error(`${e}: ${r instanceof Error?r.message:String(r)}`,{cause:r})]}}}export{a as e};
